### PR TITLE
Clear (saved) bg processor data after a dispatch

### LINKED ISSUE #1602: process_course_data arguments in multiple llms_background_process_course_data_batch rows
### Reproduction Steps

I'm not sure of the steps to reproduce this issue.

It appears that it is possible for `LLMS_Processor_Course_Data->dispatch_calc()` to add a new task with a course ID and page number to an existing batch, which is then saved as a new batch. So now there are multiple batches that contain the same task.

### Expected Behavior

When `LLMS_Processor_Course_Data->dispatch_calc()` is called, a new batch is saved containing tasks for only the given course ID.

### Actual Behavior

In the wp_options table, I have
```
llms_background_process_course_data_batch_49929a7ab6937b11bad017 = a:1:{i:0;a:5:{s:7:"post_id";s:5:"12926";s:8:"statuses";a:1:{i:0;s:8:"enrolled";}s:4:"page";i:1;s:8:"per_page";i:100;s:4:"sort";a:1:{s:2:"id";s:3:"ASC";}}}

llms_background_process_course_data_batch_2bd6ccdf7c02c959dac0d0 = a:2:{i:0;a:5:{s:7:"post_id";s:5:"12926";s:8:"statuses";a:1:{i:0;s:8:"enrolled";}s:4:"page";i:1;s:8:"per_page";i:100;s:4:"sort";a:1:{s:2:"id";s:3:"ASC";}}i:1;a:5:{s:7:"post_id";s:5:"13076";s:8:"statuses";a:1:{i:0;s:8:"enrolled";}s:4:"page";i:1;s:8:"per_page";i:100;s:4:"sort";a:1:{s:2:"id";s:3:"ASC";}}}

llms_background_process_course_data_batch_ec0b0ed1824c59b8a2e09d = a:3:{i:0;a:5:{s:7:"post_id";s:5:"12926";s:8:"statuses";a:1:{i:0;s:8:"enrolled";}s:4:"page";i:1;s:8:"per_page";i:100;s:4:"sort";a:1:{s:2:"id";s:3:"ASC";}}i:1;a:5:{s:7:"post_id";s:5:"13076";s:8:"statuses";a:1:{i:0;s:8:"enrolled";}s:4:"page";i:1;s:8:"per_page";i:100;s:4:"sort";a:1:{s:2:"id";s:3:"ASC";}}i:2;a:5:{s:7:"post_id";s:5:"13083";s:8:"statuses";a:1:{i:0;s:8:"enrolled";}s:4:"page";i:1;s:8:"per_page";i:100;s:4:"sort";a:1:{s:2:"id";s:3:"ASC";}}}
```

etc.

This issue has been recreated:
+ [x] Locally
+ [ ] On a staging site
+ [x] On two production websites
+ [ ] With only LifterLMS and a default theme

### PR DESCRIPTION
**Pull request by thomasplevy**
## Description

Fixes #1602

## How has this been tested?

+ New & existing phpunit tests

## Types of changes
Bugfix

## Checklist:
- [x] My code has been tested.
- [x] My code passes all existing automated tests. 
- [x] My code follows the LifterLMS Coding & Documentation Standards.

### CODE REVIEW

File: includes/abstracts/llms.abstract.processor.php [hunk 1]
```diff
@@ -33,29 +33,15 @@ abstract class LLMS_Abstract_Processor extends WP_Background_Process {
 
 	/**
 	 * Initializer
+	 *
 	 * Acts as a constructor that extending processors should implement
 	 * at the very least should populate the $this->actions array
```
Comment (eri-trabiccolo) [inline includes/abstracts/llms.abstract.processor.php]:
```suggestion
	 * at the very least should populate the $this->actions array.
```

File: includes/abstracts/llms.abstract.processor.php [hunk 2]
```diff
@@ -183,10 +196,9 @@ public function get_data( $key = null, $default = '' ) {
 	/**
 	 * Log data to the processors log when processors debugging is enabled
 	 *
-	 * @param    mixed $data  data to log
-	 * @return   void
-	 * @since    3.15.0
-	 * @version  3.15.0
+	 * @param mixed $data  Data to log.
```
Comment (eri-trabiccolo) [inline includes/abstracts/llms.abstract.processor.php]:
```suggestion
	 * @param mixed $data Data to log.
```

File: includes/abstracts/llms.abstract.processor.php [hunk 3]
```diff
@@ -183,10 +196,9 @@ public function get_data( $key = null, $default = '' ) {
...
+	 * @return void
+	 * @since 3.15.0
```
Comment (eri-trabiccolo) [inline includes/abstracts/llms.abstract.processor.php:199]: mmm this should be moved back above

**Review by eri-trabiccolo — CHANGES_REQUESTED** [review]
docs :P

File: includes/abstracts/llms.abstract.processor.php (hunk 3 above)
Comment (thomasplevy) [inline includes/abstracts/llms.abstract.processor.php]:
```suggestion
	 * @since 3.15.0
	 *
	 * @param mixed $data Data to log.
	 * @return void
```

**Pull request merged**